Modules/_functoolsmodule.c

```diff
@@ -1194,6 +1189,13 @@ lru_cache_new(PyTypeObject *type, PyObject *args, PyObject *kw)
         return NULL;
     }
 
+    state = get_functools_state_by_type(Py_TYPE(obj));
+    if (state == NULL) {
+        Py_DECREF(cachedict);
+        Py_DECREF(obj);
```

Comment:
Hum. I am not sure if it's safe to call lru_cache_dealloc() here, obj is not fully initialized. For example, lru_cache_unlink_list() can crash, no? Maybe call get_functools_state_by_type(type) before creating obj?

Reply:
Done